Octokit.Tests/Clients/BlobClientTests.cs

```diff
@@ -122,7 +122,7 @@ public void CanDeserializeBlob()
             Assert.NotNull(response.Body);
             Assert.Equal(blobResponseJson, (string)response.HttpResponse.Body);
             Assert.Equal(100, response.Body.Size);
-            Assert.Equal(EncodingType.Utf8, response.Body.Encoding);
+            Assert.Equal(EncodingType.Utf8, response.Body.Encoding.ParsedValue);
```

Comment:
It seems it is currently, eventhough I agree the implicit conversion should mean it's not.  

Ive traced the reason:  

`EncodingType.Utf8` corresponds to an api value of `utf-8` but our response enum doesn't include a custom `[Property()]` attribute on the members.  We still deserialize the API value OK, since the deserializer removes underscores and hyphens... but when we are converting this back the other way (ie with the changes I made in `StringEnum`'s `ctor` we have no way to know that `EncodingType.Utf8` should be `utf-8`.  

To fix this, of course we could add the attribute to this enum, but there is nothing that stops this situation from existing on any other enum value that is only used in API responses.  When an enum is used in a request we HAVE to know what the correct api string value is, but when only used in responses we dont currently (need to) know the value.

One possibility would be to enforce (via convention test) that all enums (both request and response) have custom `Property()` attributes set.  Although this could be a fair few edits, it is the most explicit and obvious approach.  It would also let us remove the "magic" removal of hypens and underscores in the deserializer, instead only ever relying on explicit parameter mapping attributes.  Thoughts?